### PR TITLE
Feature-complete commandline interface

### PR DESCRIPTION
**Pull request by sparkprime**
@sbarzowski @lwander

### CODE REVIEW

**Comment by sparkprime** [comment]
```
dcunnin@dcunnin7:~/go-jsonnet$ ./tests.sh 
snippet: true
examples/check.sh: All 14 tests executed correctly.
examples/terraform/check.sh: All 4 tests executed correctly.
test_suite/run_tests.sh: All 51 test scripts pass.
```

File: jsonnet/cmd.go [hunk 1]
```diff
@@ -45,6 +168,322 @@ func getVar(s string) (string, string, error) {
...
+	if len(remainingArgs) == 0 {
+		usage(os.Stderr)
+		return false, fmt.Errorf("ERROR: Must give %s", want)
+	}
+
+	// TODO(dcunnin): Formatter allows multiple files in test and in-place mode.
+	multipleFilesAllowed := false
+
+	if !multipleFilesAllowed {
+		filename := remainingArgs[0]
```
Comment (lwander) [inline jsonnet/cmd.go]: Why assign this out here? It's only used in the nested error block below once.

File: jsonnet/cmd.go [hunk 2]
```diff
@@ -45,6 +168,322 @@ func getVar(s string) (string, string, error) {
...
+		if len(remainingArgs) > 1 {
+			usage(os.Stderr)
+			return false, fmt.Errorf("ERROR: Already specified %s as \"%s", want, filename)
```
Comment (lwander) [inline jsonnet/cmd.go]: The error feels cryptic when all you're disallowing is multiple files to be specified.

File: jsonnet/cmd.go [hunk 3]
```diff
@@ -45,6 +168,322 @@ func getVar(s string) (string, string, error) {
...
+			return err
+		}
+	}
+
+	return nil
+}
+
+// writeOutputStream writes the output as a YAML stream.
+func writeOutputStream(output []string, outputFile string) error {
+
```
Comment (lwander) [inline jsonnet/cmd.go]: Extra newline

**Comment by sparkprime** [comment]
Need a way to get coverage numbers to include the centralized tests.

**Pull request merged**